### PR TITLE
Clean mdo schemas

### PR DESCRIPTION
**Pull request by gnodet**
* fix namespace location
* remove unneeded CDATA
* code formatting
* use `{@code xxx}` and `@see`
* generate location tracking classes
* add location tracking to toolchain model

### CODE REVIEW

**Review by cstamas — APPROVED** [review]

File: api/maven-api-plugin/src/main/mdo/lifecycle.mdo [hunk 1]
```diff
@@ -18,20 +18,20 @@ under the License.
...
     Configuration of custom lifecycle mappings for the plugin, as generally stored in
-    <code>META-INF/maven/lifecycle.xml</code> in a plugin's jar artifact.
+    {@code META-INF/maven/lifecycle.xml} in a plugin's jar artifact.
   ]]></description>
   <classes>
     <class rootElement="true" xml.tagName="lifecycles" xsd.compositor="sequence">
       <name>LifecycleConfiguration</name>
       <version>1.0.0</version>
-      <description><![CDATA[Root element of the <code>lifecycle.xml</code> file.]]></description>
+      <description><![CDATA[Root element of the {@code lifecycle.xml} file.]]></description>
```
Comment (elharo) [inline api/maven-api-plugin/src/main/mdo/lifecycle.mdo:34]: CDATA no longer needed here

File: api/maven-api-model/src/main/mdo/maven.mdo [hunk 2]
```diff
@@ -572,15 +550,12 @@
...
-            This element describes all of the dependencies associated with a
-            project.
+            This element describes all the dependencies associated with a project.
             These dependencies are used to construct a classpath for your
             project during the build process. They are automatically downloaded from the
             repositories defined in this project.
-            See <a href="https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html">the
-            dependency mechanism</a> for more information.
-            ]]>
+            See &lt;a href="https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html">the
```
Comment (michael-o) [inline api/maven-api-model/src/main/mdo/maven.mdo:557]: `@see`?

Comment (gnodet) [reply api/maven-api-model/src/main/mdo/maven.mdo:557]: Isn't `@see` to link to another javadoc element ?

Comment (gnodet) [reply api/maven-api-model/src/main/mdo/maven.mdo:557]: Ah, https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#see

Comment (gnodet) [reply api/maven-api-model/src/main/mdo/maven.mdo]: Fixed

**Pull request merged**